packages/graphback-cli/src/helpers/generate.ts

```diff
@@ -43,7 +45,7 @@ export async function generateBackend(): Promise<void> {
 
     const backend: GraphQLBackendCreator = new GraphQLBackendCreator(schemaText, generation)
 
-    const generated: IGraphQLBackend = await backend.createBackend(database)
+    const generated: IGraphQLBackend = await backend.createBackend(database, templateType)
```

Comment:
-1 for template engine to be aware of the templates. This introduces complexity and kinda breaks the onion architecture of the solution.

If we generate resolvers in some specific way that have all the info and then just map output format directl in template we going to prove extensivility.

Developers should be able to create tons of templates without changing CLI or core of graphback.

Reply:
But then it will output according to what the template needs right ?

Reply:
Yeah, but think about complexity that this is putting. 
There could be many templates that we will support with different variations. This is also making things quite hard to test - every change in generation needs to be tested in 2 variations. 


I think we can simply generate resolvers in format that provides a lots of metadata then we can adjust that to new format directly in template. 

For example:
```
{ Queries: { find() }, Mutations: {}
```

can be transformed to
```
{ ...Queries, ...Mutations} 
```

Reply:
Hmm we are doing that only, we generate metadata using `targetResolverContext` and inject them to `apollo-template` and `graphqljs-template`, respectively.

Reply:
I removed the bit where we were using template info inside metadata generation in newest commit.